README.md

```diff
@@ -31,6 +33,292 @@ composer require yiisoft/yii-middleware --prefer-dist
 
 ## General usage
 
+All classes are separate implementations of [PSR 15](https://github.com/php-fig/http-server-middleware)
+middleware and do not interact with each other in any way.
+
+### `BasicNetworkResolver`
+
+Updates an instance of server request with protocol from special headers.
+
+It can be used in the following cases:
+
+- not required IP resolve to access the user's IP
+- user's IP is already resolved (eg `ngx_http_realip_module` or similar)
```

Comment:
These are not specifically clear. Usually it is used when you work behind a load balancer. User is getting HTTPS but all the internal network uses HTTP.